students/yushus/session03/list_lab.py

```diff
@@ -0,0 +1,84 @@
+def series_1():
+    fruits = ['Apples', 'Pears', 'Oranges', 'Peaches']
+    print('Initial fruit list:')
+    print(fruits)
+
+    new_fruit = input('Please enter your favorite fruit: ')
+    fruits.append(new_fruit.strip())
+    print('The new fruit list with your favorite one included at last:')
+    print(fruits)
+
+    num = input('Please tell me what fruit you like the most, use a number: ')
+    index = int(num.strip())-1
+    if 0 <= index < len(fruits):
+        print('Wow, your favorite fruit is {num} {fruit}'.format(num=num, fruit=fruits[index]))
+    else:
+        print("Fruit not in the list. Sorry...")
+
+    add_fruit = ['kiwi']
+    fruits = add_fruit + fruits
+    print('A new secret fruit just got added in front: ') 
+    print('Here is all the fruits now:') 
+    print(fruits)
+
+    insertFront_fruit = 'dragonfruit'
+    fruits.insert(0,insertFront_fruit)
+    print('Another secret fruit got added in front: ')
+    print(fruits)
+
+    print('Fruits start with \'P\': ')
+    for i in fruits:
+        if i.startswith('P'):
+            print(i)
+
+def series_2():
+    fruits = ['Apples', 'Pears', 'Oranges', 'Peaches']
+    print(fruits)
+
+    fruits.remove(fruits[-1])
+    print(fruits)
+
+    fruit_to_delete = input("Please choose a fruit to delete: ")
+    for f in fruits:
+        if f.lower() == fruit_to_delete.strip().lower():
+            fruits.remove(f)
+    print(fruits)
+
+    new_fruits = fruits + fruits
+    found = False
+    while(not found):
+        fruit_to_delete = input("Please try again: ").strip()
+        for f in new_fruits:
+            if f.lower() == fruit_to_delete.lower():
+                new_fruits.remove(f)
```

Comment:
You are modifying the `new_fruits` list here, and yet you are iterating over the `new_fruits` list in line 51. Because the list may be modified (some items removed) in the loop, the loop's behavior might be unpredictable. Better approach is to iterate over a copy of `new_fruits` in line 51: `for f in new_fruits[:]:'